webapp/app.py

```
@@ -712,6 +764,95 @@ def _build_key_diacritic_hints(self):
+    Returns None if no definition found.
+    """
+    cache_dir = os.path.join(WORD_DEFS_DIR, lang_code)
+    cache_path = os.path.join(cache_dir, f"{word.lower()}.json")
+
+    # Check cache first
+    if os.path.exists(cache_path):
+        try:
+            with open(cache_path, "r") as f:
+                return json.load(f)
+        except Exception:
+            pass
```

Ruff (0.15.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)